server/parameters/cli.rs

```diff
@@ -29,4 +30,22 @@ pub struct CLIArgs {
     /// Path to the data directory
     #[arg(long = "storage.data", value_name = "DIR")]
     pub storage_data: Option<String>,
+
+    /// Enable usage metrics reporting
+    #[arg(long = "diagnostics.reporting.metrics", default_value_t = true, action=ArgAction::Set)]
+    pub diagnostics_reporting_metrics: bool, // used to be `statistics` in 2.x
```

Comment:
Called it `metrics` as we will have `errors` (for sentry). At the same time, `PostHog` should not be configured by a separate flag, so something general should go fine. 
Diagnostics generally consist of metrics in our model, so it's... bearable.